open-craft/frontend-app-course-authoring
# [WIP] [FC-0036] Search Content Tags

## Description

This PR adds the the ability to search tags in the Content Tags Drawer.

![Screen Shot 2023-12-04 at 1 49 05 PM](https://github.com/open-craft/frontend-app-course-authoring/assets/6829768/0cbb6077-042d-47e2-9e22-fe64de075ca1)

## Supporting information

Related Tickets:
- https://github.com/openedx/modular-learning/issues/122
- Depends on https://github.com/openedx/frontend-app-course-authoring/pull/704

## Testing instructions

1. Run you local dev stack.
2. Next, navigate to http://localhost:18010/admin/waffle/flag/ and create a new flag: new_studio_mfe.use_tagging_taxonomy_list_page make sure it is active for everyone
3. If you don't already have sample taxonomy/tags data, follow the instructions in this repo to generate sample data: https://github.com/open-craft/taxonomy-sample-data
4. Navigate to the sample courses and open the tags drawer for the units
5. Check that all the taxonomies are shown in the drawer and the applied tags are shown
6. Expand some taxonomies and click on the "Add Tags" button
7. In the search bar, type in a term to search for in the tags
8. Confirm that the search results contain the search term. If the root tags shown do not contain the search term, that means the term is in one its (grand)children
9. Confirm that when you check/uncheck a tag it is reflected in the tags tree
10. Confirm that if you check a tag, all its ancestors become implicit
11. Confirm when you uncheck a tag and it is the only child, all its ancestors get unchecked
12. Confirm that the tags number gets updated correctly whenever you add/remove tags
---
Private-ref: [FAL-3572](https://tasks.opencraft.com/browse/FAL-3572)
